packages/plugin-legacy/src/index.ts

```diff
@@ -344,7 +344,8 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
             fileName = fileName.replace(/[.-]?\[hash\]/, '-legacy$&')
           } else {
             // entry.js -> entry-legacy.js
-            fileName = fileName.replace(/(.+)\.(.+)/, '$1-legacy.$2')
+            // entry.min.js -> entry-legacy.min.js
+            fileName = fileName.replace(/(.+?)\.(.+)/, '$1-legacy.$2')
```

Comment:
This is a mini breaking change as before `entry.min.js` would be `entry.min-legacy.js`, but luckily we were also fixing this for plugin-legacy's next major: https://github.com/vitejs/vite/pull/11599